### PR TITLE
Lease controller - Delete

### PR DESCRIPTION
**Pull request by AmudaPalani**
## Proposed changes

Use new data model for leases Delete controller. This is a soft delete, meaning change lease status from Active to Inactive. 

The dynamodb stream from Lease change will take care of updating Account status to "Not ready".

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (changes to code, which do not change application behavior)

## Checklist

- [ ] I have filled out this PR template
- [ ] I have read the [CONTRIBUTING](https://github.com/Optum/dce/blob/master/docs/CONTRIBUTING.md) doc
- [ ] I have added automated tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (`README.md`, inline comments, etc.)
- [ ] I have updated the `CHANGELOG.md` under a `## next` release, with a short summary of my changes

### CODE REVIEW

File: pkg/data/dataiface/lease.go [hunk 1]
```diff
@@ -12,9 +12,6 @@ type LeaseData interface {
 	// Get the Lease record by ID
 	Get(ID string) (*lease.Lease, error)
 
-	// Delete the Lease record in DynamoDB
```
Comment (kddejong) [inline pkg/data/dataiface/lease.go]: 👍

File: cmd/lambda/leases/delete.go [hunk 2]
```diff
@@ -3,119 +3,65 @@ package main
...
+		api.WriteAPIErrorResponse(w, err)
 		return
 	}
 
-	leaseResponse := response.LeaseResponse(*lease)
-	err = json.NewEncoder(w).Encode(leaseResponse)
-	if err != nil {
-		response.WriteServerErrorWithResponse(w, fmt.Sprintf("Failed Decommission on Account Lease %q", leaseID))
+	if len(*leases) == 0 {
+		response.WriteRequestValidationError(w, fmt.Sprintf("No leases found for Principal %q and Account ID %q", *queryLease.PrincipalID, *queryLease.AccountID))
```
[the remark this answers is not in the document]
Comment (AmudaPalani) [reply cmd/lambda/leases/delete.go]: @kddejong Do we want the DeleteLease query to have principalID and accountID?

File: tests/acceptance/api_test.go [hunk 3]
```diff
@@ -700,7 +700,7 @@ func TestApi(t *testing.T) {
 					// Verify error response json
 					// Get nested json in response json
 					errResp := data["error"].(map[string]interface{})
-					assert.Equal(r, "ClientError", errResp["code"].(string))
+					assert.Equal(r, "RequestValidationError", errResp["code"].(string))
```
Comment (eschwartz) [inline tests/acceptance/api_test.go]: What would you think about making this a `404 Not Found` error instead?
The client is requesting to delete a specific resource, and that resource doesn't exist -- I would usually consider that a 404.

Comment (AmudaPalani) [reply tests/acceptance/api_test.go]: Fixed

**Pull request merged**